jetcd-core/src/main/java/com/coreos/jetcd/Watch.java

```diff
@@ -48,11 +48,12 @@
   /**
    * Interface of Watcher.
    */
-  interface Watcher {
+  interface Watcher extends AutoCloseable {
```

Comment:
why `extends` not `implement` the  `AutoCloseable`?

if we `@Override` the `close()` from `AutoCloseable`, how come our `close()` doesn't `throw Exception`?

Reply:
1. `interface` only can 'extends' another interface. not `implement`.

2. The overriding method can throw those checked exceptions, which have less scope than the exception(s) declared in the overridden method.

Reply:
now i understand what you were doing. pretty cool java trick.